From: Jeff Layton <jlayton@kernel.org>
Date: Tue, 26 Jan 2021 11:49:54 -0500
Subject: ceph: only check pool permissions for regular files
Git-commit: e9b2250156c381b0973ea6ec3890fe8706426ecc
Patch-mainline: v5.13-rc1
References: bsc#1186501

There is no need to do a ceph_pool_perm_check() on anything that isn't a
regular file, as the MDS is what handles talking to the OSD in those
cases. Just return 0 if it's not a regular file.

Reported-by: Luis Henriques <lhenriques@suse.de>
Signed-off-by: Jeff Layton <jlayton@kernel.org>
Reviewed-by: Xiubo Li <xiubli@redhat.com>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>

---
 fs/ceph/addr.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/fs/ceph/addr.c b/fs/ceph/addr.c
index bc9864524fde..aa612bc8a559 100644
--- a/fs/ceph/addr.c
+++ b/fs/ceph/addr.c
@@ -1941,6 +1941,10 @@ int ceph_pool_perm_check(struct inode *inode, int need)
 	s64 pool;
 	int ret, flags;
 
+	/* Only need to do this for regular files */
+	if (!S_ISREG(inode->i_mode))
+		return 0;
+
 	if (ci->i_vino.snap != CEPH_NOSNAP) {
 		/*
 		 * Pool permission check needs to write to the first object.

